pkgs/by-name/po/positron-bin/package.nix

```
@@ -0,0 +1,139 @@
+{
+  lib,
+  _7zz,
+  alsa-lib,
+  systemd,
+  autoPatchelfHook,
+  blas,
+  dpkg,
+  fetchurl,
+  gtk3,
+  libglvnd,
+  libxkbcommon,
+  makeShellWrapper,
+  mesa,
+  musl,
+  nss,
+  patchelf,
+  stdenv,
+  xorg,
+}:
+let
+  pname = "positron-bin";
+  version = "2024.11.0-116";
+in
+stdenv.mkDerivation {
+  inherit version pname;
+
+  src =
+    if (stdenv.isDarwin) then
+      fetchurl {
+        url = "https://github.com/posit-dev/positron/releases/download/${version}/Positron-${version}.dmg";
+        hash = "sha256-5Ym42InDgFLGdZk0LYV1H0eC5WzmsYToG1KLdiGgTto=";
+      }
+    else
+      fetchurl {
+        url = "https://github.com/posit-dev/positron/releases/download/${version}/Positron-${version}.deb";
+        hash = "sha256-pE25XVYFW8WwyQ7zmox2mmXy6ZCSaXk2gSnPimg7xtU=";
+      };
+
+  buildInputs =
+    [ makeShellWrapper ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      alsa-lib
+      dpkg
+      gtk3
+      libglvnd
+      libxkbcommon
+      mesa
+      musl
+      nss
+      stdenv.cc.cc
+      xorg.libX11
+      xorg.libXcomposite
+      xorg.libXdamage
+      xorg.libxkbfile
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      blas
+      patchelf
+    ];
+
+  nativeBuildInputs =
+    lib.optionals stdenv.hostPlatform.isLinux [
+      autoPatchelfHook
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      _7zz
+    ];
+
+  runtimeDependencies = lib.optionals stdenv.hostPlatform.isLinux [
+    # Needed to fix the "Zygote could not fork" error.
+    (lib.getLib systemd)
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    ${builtins.toString (
+      lib.optionals stdenv.hostPlatform.isLinux ''dpkg-deb --fsys-tarfile "$src" | tar -x --no-same-owner''
+    )}
```

```suggestion
    ${lib.optionalString stdenv.hostPlatform.isLinux ''dpkg-deb --fsys-tarfile "$src" | tar -x --no-same-owner''}
```
This might resolve the list to string conversion error.

---

Yes, that works too and looks much better than `builtins.toString`. Thanks!